### PR TITLE
Re-enable CUDA compiler and runtime build.

### PR DESCRIPTION
**Pull request by stellaraccident**
* Both are now disabled by default. Enabling them triggers a search for headers and bitcode files that finds them in a local SDK or allows them to be externally specified.
* Removes bazel build support for CUDA runtime.
* Does not yet re-enable tests. Will clean that up in a followup as there is now some optionality here that was not planned for.

### CODE REVIEW

**Review by ThomasRaoux — APPROVED** [review]
Thanks Stella!

File: CMakeLists.txt [hunk 1]
```diff
@@ -323,6 +319,48 @@ if(IREE_ERROR_ON_MISSING_SUBMODULES AND Python3_FOUND AND Git_FOUND)
...
+# messaging to do this all in one place, since we can provide very targeted
+# advice.
+#-------------------------------------------------------------------------------
+
+set(IREE_CUDA_LIBDEVICE_PATH "" CACHE STRING "Absolute path to an appropriate libdevice.*.bc (needed to build the IREE cuda compiler target)")
+
+# If any CUDA features are being built, try to locate a CUDA SDK. We will fall
+# back to this as needed for specific features.
+if(IREE_TARGET_BACKEND_CUDA OR IREE_HAL_DRIVER_CUDA)
+  find_package(CUDAToolkit QUIET)
```
Comment (ScottTodd) [inline CMakeLists.txt:335]: note that we could put a version in here: https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#arguments

(if we know that we need at least a specific version, or one exact version)

File: CMakeLists.txt [hunk 2]
```diff
@@ -323,6 +319,48 @@ if(IREE_ERROR_ON_MISSING_SUBMODULES AND Python3_FOUND AND Git_FOUND)
...
+
+  if(EXISTS "${IREE_CUDA_LIBDEVICE_PATH}")
+    message(STATUS "Using CUDA libdevice: ${IREE_CUDA_LIBDEVICE_PATH}")
+  else()
+    message(SEND_ERROR "Cannot find CUDA libdevice file (${IREE_CUDA_LIBDEVICE_PATH}). Either configure your CUDA SDK such that it can be found or specify explicitly via -DIREE_CUDA_LIBDEVICE_PATH=/path/to/libdevice.10.bc")
+  endif()
+endif()
+
+if(IREE_HAL_DRIVER_CUDA AND NOT CUDAToolkit_FOUND)
+  message(SEND_ERROR "Cannot build IREE runtime CUDA components (-DIREE_HAL_DRIVER_CUDA=ON) because a CUDA SDK was not found. Consult CMake docs for your version: https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html")
```
Comment (ScottTodd) [inline CMakeLists.txt:361]: Thanks for the useful error message :)

**Review by ScottTodd — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (stellaraccident) [reply CMakeLists.txt:335]: Yes, I don't actually know such things so will leave it to the more informed to make a decision.

File: CMakeLists.txt (hunk 2 above)
Comment (stellaraccident) [reply CMakeLists.txt:361]: I like helpful error messages.

**Pull request merged**